### ISSUE TITLE
How can ApolloModule be used to communicate with Multiple GraphQL endpoints?

### ISSUE THREAD

**Issue by chrishandorf** [apollostack/apollo-angular#246]
ApolloModule provides the withClient() function to define an ApolloClient and its associated single NetworkInterface (URI).  How can this module be used to talk to multiple GraphQL endpoints from a single application?

**Comment by kamilkisiela** [comment]
I was thinking about introducing named clients.

```ts
type ClientMap = {[name: string]: ApolloClient};

function provideClients(): ClientMap {
  return {
    main: mainApolloClient,
    extra: extraApolloClient,
  };
}

ApolloModule.withClients(provideClients);

class AppComponent {
  constructor(private apollo: Angular2Apollo) {}

  ngOnInit() {
    this.feed = this.apollo.use('main').watchQuery(...);
  }
}
```

Do you have a simpler idea?

**Comment by kamilkisiela** [comment]
#263

**Comment by chrishandorf** [comment]
This looks great.  I was thinking the same type of interface.  Nice work!

**Comment by kamilkisiela** [comment]
It's going to be released with `v0.10.0`, expect it to happen in few days.
